src/finance/PoolSteward.sol

```diff
@@ -0,0 +1,191 @@
+        address pool,
+        address reserve,
+        uint256 amount
+    ) external onlyOwnerOrGuardian {
+        if (amount == 0) revert InvalidZeroAmount();
+
+        CU.IOInput memory withdrawData = CU.IOInput(pool, reserve, amount);
+
+        CU.withdrawFromV3(COLLECTOR, withdrawData, address(COLLECTOR));
+    }
+
+    /// @inheritdoc IPoolSteward
+    function withdrawV2(
+        address pool,
+        address reserve,
+        uint256 amount
+    ) external onlyOwnerOrGuardian {
+        if (amount == 0) revert InvalidZeroAmount();
+
+        _validateV2Pool(pool);
+
+        CU.IOInput memory withdrawData = CU.IOInput(pool, reserve, amount);
+        CU.withdrawFromV2(COLLECTOR, withdrawData, address(COLLECTOR));
+    }
+
+    /// @inheritdoc IPoolSteward
+    function migrateBetweenV3(
+        address fromPool,
+        address toPool,
+        address reserve,
+        uint256 amount
+    ) external onlyOwnerOrGuardian {
+        if (amount == 0) revert InvalidZeroAmount();
+
+        _validateV3Pool(toPool);
+
+        CU.IOInput memory withdrawData = CU.IOInput(fromPool, reserve, amount);
+        uint256 withdrawnAmt = CU.withdrawFromV3(
+            COLLECTOR,
+            withdrawData,
+            address(COLLECTOR)
+        );
+
+        CU.IOInput memory depositData = CU.IOInput(
+            toPool,
+            reserve,
+            withdrawnAmt
+        );
+        CU.depositToV3(COLLECTOR, depositData);
+    }
+
+    /// @inheritdoc IPoolSteward
+    function migrateV2toV3(
+        address v2Pool,
+        address v3Pool,
+        address reserve,
+        uint256 amount
+    ) external onlyOwnerOrGuardian {
+        if (amount == 0) revert InvalidZeroAmount();
+
+        _validateV2Pool(v2Pool);
+        _validateV3Pool(v3Pool);
+
+        CU.IOInput memory withdrawData = CU.IOInput(v2Pool, reserve, amount);
+        uint256 withdrawnAmt = CU.withdrawFromV2(
+            COLLECTOR,
+            withdrawData,
+            address(COLLECTOR)
+        );
+
+        CU.IOInput memory depositData = CU.IOInput(
+            v3Pool,
+            reserve,
+            withdrawnAmt
+        );
+        CU.depositToV3(COLLECTOR, depositData);
+    }
+
+    /// DAO Actions
+
+    /// @inheritdoc IPoolSteward
+    function approvePool(address newPool, bool isVersion3) external onlyOwner {
+        _approvePool(newPool, isVersion3);
+    }
+
+    /// @inheritdoc IPoolSteward
+    function revokePool(address pool, bool isVersion3) external onlyOwner {
+        _revokePool(pool, isVersion3);
+    }
+
+    /// Logic
+
+    /// @dev Internal function to approve an Aave V3 Pool instance
+    function _approvePool(address pool, bool isVersion3) internal {
+        if (pool == address(0)) revert InvalidZeroAddress();
+
+        if (isVersion3) {
+            poolsV3[pool] = true;
+        } else {
+            poolsV2[pool] = true;
+        }
+        emit ApprovedPool(pool, isVersion3 ? bytes("v3") : bytes("v2"));
```

Comment:
wouldn't it make sense to have a ApprovedV2Pool & ApprovedV3Pool?
Imo a bit cleaner & cheaper.

Same for RevokePool(which currently lacks the pool version info)